src/lsm/README.md

```diff
@@ -0,0 +1,111 @@
+# Tree
+## Compaction
+
+Tree compaction runs to the sound of music!
+
+Compacting LSM trees involves merging and moving tables into the next levels as needed.
+To avoid write amplification stalls and bound latency, compaction is done incrementally.
+
+A full compaction phase is denoted as a bar or measure, using terms from music notation.
+Each bar consists of `lsm_batch_multiple` beats or "compaction ticks" of work.
```

Comment:
When do these beats happen? Who drives them?

Reply:
Is that not explained sufficiently by the next sentence? Or do you mean at a higher level, what is driving `compact()`?

Reply:
Yeah. I get the sense that 1 committed op = 1 beat, but I don't think that's explicitly spelled out here. My understanding is that every `lsm_batch_multiple` ops a new compaction is started and runs concurrently with op commits. If true, that would be good context to start with.